packages/kbn-typed-react-router-config/src/create_router.ts

```diff
@@ -110,10 +110,13 @@ export function createRouter<TRoutes extends RouteMap>(routes: TRoutes): Router<
       throw new Error(errorMessage);
     }
 
+    const hasExactMatch = matches.some((match) => match.match.isExact);
+
     return matches.slice(0, matchIndex + 1).map((matchedRoute) => {
       const route = routesByReactRouterConfig.get(matchedRoute.route);
 
-      if (route?.params) {
+      // Only validates the params when exact match has been found, otherwise a 404 page should be show
+      if (hasExactMatch && route?.params) {
```

Comment:
this looks like a significant change - have you considered the ramifications? (I'm unsure myself)

Reply:
@dgieselaar all the params will be validated across all tree if there was an exact match for the path received, just like before. The difference now is if an invalid path is received, that doesn't match any of the ones we defined, there's no reason the validate the params of the paths up in the tree, because we'll show the 404 component instead.

example: 
`/apm/services`
- validates params from `/services`
- validates params from `/`

`/apm/services/foo`
- Does not validate any param as the 404 component will be shown.

Reply:
I don't think it should return the routes if they are not validated. What happens if we return `undefined` in that case?

Reply:
I reversed the way it was before